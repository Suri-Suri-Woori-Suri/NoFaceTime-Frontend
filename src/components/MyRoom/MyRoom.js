import React from 'react';

import RoomList from '../RoomList/RoomList';
import styles from './MyRoom.module.css';

const MyRoom = ({
  currentUser,
  enterRoom,
  popupModal,
  fetchToDeleteRoomData
}) => {
  return (
    <div className={styles.MyRoom}>
      <div className={styles.Title}>Your Room</div>
      <div className={styles.Rooms}>
        <RoomList
          currentUser={currentUser}
          enterRoom={enterRoom}
          fetchToDeleteRoomData={fetchToDeleteRoomData}
        />
      </div>
      <button
        className={styles.AddRoomButton}
        onClick={popupModal}>
        Add Room
      </button>
    </div>
  );
};

export default MyRoom;
