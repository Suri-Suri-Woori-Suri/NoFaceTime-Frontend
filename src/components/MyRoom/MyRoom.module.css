.MyRoom {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
  height: 85%;
  font-size: 20px;
  text-align: center;
  box-sizing: border-box;
}

.RoomsContainer {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 95%;
  height: 80%;
  border: 1px solid #e5eaf5;
  border-radius: 10px;
  font-weight: 250;
  background-color: whitesmoke;
  background-color: #e5eaf5;
  box-sizing: border-box;
  overflow: scroll;
  padding-top: 50px;
}

.Title {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 95%;
  height: 10%;
  font-size: 2em;
  margin: 5px;
  box-sizing: border-box;
}

.RoomNameInput {
  /*?*/
  width: 400px;
  height: 40px;
  border: 1px solid #c9d3dd;
  background: red;
  border-radius: 10px;
  text-align: center;
}

.AddRoomButton {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 150px;
  height: 40px;
  padding: 0px 10px 0px 10px;
  background: #274ba0;
  color: #ffffff;
  border-color: #274ba0;
  border-width: 1px;
  border-style: solid;
  border-radius: 20px 20px 20px 20px;
  font-family: "Oswald";
  font-weight: 600;
  font-size: 1rem;
  text-align: center;
}

.AddRoomButton:hover {
  transform: translateY(2px);
}
