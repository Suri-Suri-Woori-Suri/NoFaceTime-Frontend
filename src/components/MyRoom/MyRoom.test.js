import React from 'react';
import { mount } from 'enzyme';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import MyRoom from './MyRoom';

describe('<MyRoom/>', () => {
  afterEach(cleanup);

  const mockCurrentUser = [
    {
      name: 'group1',
      _id: 'a',
    },
    {
      name: 'group2',
      _id: 'b',
    }
  ];
  const mockEnterRoom = jest.fn();
  const mockPopupModal = jest.fn();
  const mockCheckBox = jest.fn();

  const wrapper = mount(
    <MyRoom
      currentUser={mockCurrentUser}
      enterRoom={mockEnterRoom}
      popupModal={mockPopupModal}
      handleCheckbox={mockCheckBox}
    />
  );

  it('should be render add room button correctly', () => {
    expect(wrapper.containsMatchingElement('Add Room')).toEqual(true);
  });

  it('should be invoked when button is clicked', () => {
    render(
      <MyRoom
        currentUser={mockCurrentUser}
        enterRoom={mockEnterRoom}
        popupModal={mockPopupModal}
        handleCheckbox={mockCheckBox}
      />
    );

    fireEvent.click(screen.getByText('Add Room'));

    expect(mockPopupModal).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
