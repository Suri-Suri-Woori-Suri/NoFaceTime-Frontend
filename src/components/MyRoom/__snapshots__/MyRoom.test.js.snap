// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyRoom/> matches snapshot 1`] = `
<MyRoom
  currentUser={
    Array [
      Object {
        "_id": "a",
        "name": "group1",
      },
      Object {
        "_id": "b",
        "name": "group2",
      },
    ]
  }
  enterRoom={[MockFunction]}
  handleCheckbox={[MockFunction]}
  popupModal={[MockFunction]}
>
  <div
    className="MyRoom"
  >
    <div
      className="RoomsContainer"
    >
      <RoomList
        currentUser={
          Array [
            Object {
              "_id": "a",
              "name": "group1",
            },
            Object {
              "_id": "b",
              "name": "group2",
            },
          ]
        }
        enterRoom={[MockFunction]}
      />
    </div>
    <button
      className="AddRoomButton"
      onClick={[MockFunction]}
    >
      Add Room
    </button>
  </div>
</MyRoom>
`;
